CONTRIBUTING.md

```diff
@@ -73,226 +73,126 @@ We'll lay out three examples below (one that uses our format for messages, one t
 3. Functions should not be set as fields for a test case (this usually means that the table-driven approach is being sidestepped and that the logic in the function should probably be factored out to cover multiple/all test cases)
 4. Avoid verbosity by creating local variables instead of constantly referring to struct field (e.g. doing `lockupKeeper := suite.App.LockupKeeper` instead of using `suite.App.LockupKeeper` every time).
 
-### Example #1: [Message-Related Test](https://github.com/osmosis-labs/osmosis/blob/f0270d04bd77cc5e1c23f7913118b3c2ba737e97/x/tokenfactory/keeper/admins_test.go#L122-L181) 
+### Example #1: [Message-Related Test]
 This type of test is mainly for functions that would be triggered by incoming messages (we interact directly with the message server since all other metadata is stripped from a message by the time it hits the msg_server):
 
 ```go
-func (suite *KeeperTestSuite) TestBurnDenom() {
-    var addr0bal int64
-
-    // Create a denom.
-    suite.CreateDefaultDenom()
-
-    // mint 10 default token for testAcc[0]
-    suite.msgServer.Mint(sdk.WrapSDKContext(suite.Ctx), types.NewMsgMint(suite.TestAccs[0].String(), sdk.NewInt64Coin(suite.defaultDenom, 10)))
-    addr0bal += 10
-
-    for _, tc := range []struct {
-        desc      string
-        amount    int64
-        burnDenom string
-        admin     string
-        valid     bool
-    }{
-        {
-            desc:      "denom does not exist",
-            amount:    10,
-            burnDenom: "factory/osmo1t7egva48prqmzl59x5ngv4zx0dtrwewc9m7z44/evmos",
-            admin:     suite.TestAccs[0].String(),
-            valid:     false,
-        },
-        {
-            desc:      "burn is not by the admin",
-            amount:    10,
-            burnDenom: suite.defaultDenom,
-            admin:     suite.TestAccs[1].String(),
-            valid:     false,
-        },
-        {
-            desc:      "burn amount is bigger than minted amount",
-            amount:    1000,
-            burnDenom: suite.defaultDenom,
-            admin:     suite.TestAccs[1].String(),
-            valid:     false,
+func(suite *KeeperTestSuite) TestCreateDenom() {
+    testCases := map[string] struct {
+        setup       func()
+        subdenom    string
+        valid       bool
+    } {
+
+        "subdenom too long": {
+            subdenom:   "assadsadsadasdasdsadsadsadsadsadsadsklkadaskkkdasdasedskhanhassyeunganassfnlksdflksafjlkasd",
+            valid:      false,
         },
-        {
-            desc:      "success case",
-            amount:    10,
-            burnDenom: suite.defaultDenom,
-            admin:     suite.TestAccs[0].String(),
-            valid:     true,
+        "subdenom and creator pair already exists": {
+            setup: func() {
```

Comment:
I don't think we would like to encourage folks to do setup via callbacks in test case struct definitions. Ideally, I think it's best to only keep the simplest fields in the test case struct such as strings and ints.

Reply:
callbacks are explicitly bad

Reply:
Realized that setup is not really necessary, so i removed it. Want to make the test examples very easy to follow